### ISSUE TITLE
Search requests now return 204 for no matching results

### ISSUE THREAD

**Issue by dzelemba** [GoogleCloudPlatform/healthcare-api-dicomweb-cli#44]
The tool should treat all 2XX codes as success.

E.g.

```
dcmweb   https://staging-healthcare.sandbox.googleapis.com/v1/projects/dvet-test/locations/us-central1/datasets/1/dicomStores/1/dicomWeb retrieve

Output:
/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/google/auth/_default.py:69: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK. We recommend that most server applications use service accounts instead. If your application continues to use end user credentials from Cloud SDK, you might receive a "quota exceeded" or "API not enabled" error. For more information about service accounts, see https://cloud.google.com/docs/authentication/
  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)
2020-03-16 16:15:49,923 -- Saving files into ./
Traceback (most recent call last):
  File "/usr/local/google/home/dvet/.local/bin/dcmweb", line 10, in <module>
    sys.exit(main())
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/dcmweb/command_line.py", line 66, in main
    fire.Fire(host_wrapper)
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/fire/core.py", line 138, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/fire/core.py", line 471, in _Fire
    target=component.__name__)
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/fire/core.py", line 675, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/dcmweb/dcmweb.py", line 127, in retrieve
    ids, output, type), self.multithreading)
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/dcmweb/dcmweb.py", line 35, in execute_file_transfer_futures
    for future_arguments in futures_arguments:
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/dcmweb/dcmweb.py", line 157, in _files_to_download
    .format(resources.STUDY_TAG, resources.SERIES_TAG), page)
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/dcmweb/requests_util.py", line 173, in search_instances_by_page
    + "&offset={}".format(limit*page), {}).text
  File "/usr/local/google/home/dvet/.local/lib/python3.7/site-packages/dcmweb/requests_util.py", line 128, in request
    response.status_code, response.text))
dcmweb.requests_util.NetworkError: Unexpected return code 204
```

**Comment by pavertomato** [comment]
healthcare.googleapis.com/v1beta1
still returns 200 with "[]" body for search with no matching results, should i use some different endpoint to test fix?
i can't enable staging-healthcare.sandbox.googleapis.com in console.developers.google.com
and healthcare.googleapis.com/v1beta1 returns 404 on dataset creation request

**Comment by dzelemba** [comment]
Ah yes, there are no public endpoints with this behavior just yet. I'm fine with just adding some unit tests for now and then I can do a manual test to make sure this is fixed.

**Comment by dzelemba** [comment]
Manually tested that this is fixed in v1.0.1.
